src/opentimelineio/algo/editAlgorithm.h

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Contributors to the OpenTimelineIO project
+
+#pragma once
+
+#include "opentimelineio/composition.h"
+
+namespace opentimelineio { namespace OPENTIMELINEIO_VERSION { namespace algo {
+
+//! Enum used by 3/4 Point Edit (aka. as fill)
+enum class ReferencePoint
+{
+    Source,
+    Sequence,
+    Fit
+};
+
+// Overwrite an item or items.
+//
+// | A | B |  ->  |A| C |B|
+//   ^   ^
+//   | C |
+//
+//          item = item to overwrite (usually a clip)
+//   composition = usually a track item.
+//         range = time range to overwrite.
+// remove_transitions = whether to remove transitions within range.
+// fill_template = item to fill in (usually a gap),
+//                 when range > composition's time.
+void overwrite(
+    Item*            item,
+    Composition*     composition,
+    TimeRange const& range,
+    bool             remove_transitions = true,
+    Item*            fill_template      = nullptr,
+    ErrorStatus*     error_status       = nullptr);
+
+// Insert an item.
+// |     A     | B |  ->  | A | C | A | B |
+//       ^
+//     | C |
+//
+//          item = item to insert (usually a clip)
```

Comment:
I assume that if length of AB is L1, and length of C is L2, then length of the result is L1+L2? In other words A is split?